0.download_data/scripts/3.process_videos.py

```diff
@@ -0,0 +1,310 @@
+            shutil.move(dir, new_dir_path)
+    else:
+        print(f"Unrecognized file type: {files[0].suffix}")
+
+
+# ## Convert the videos to images
+
+# In[9]:
+
+
+if run_convert_videos:
+    # set the path to the directory containing the files
+    all_videos_Path = pathlib.Path(
+        "/home/lippincm/Desktop/18TB/timelapse_data_landscapes/raw_videos/"
+    ).resolve(strict=True)
+    # ouput directory for the tiff files of frames
+    output_dir = pathlib.Path(
+        "/home/lippincm/Desktop/18TB/timelapse_data_landscapes/tiff_frames/"
+    ).resolve()
+    output_dir.mkdir(parents=True, exist_ok=True)
+    # get a list of all the files in the directory
+    list_of_files = get_files_from_dir(all_videos_Path)
+    list_of_files = unnest_list(list_of_files)
+    print(len(list_of_files))
+    list_of_files = [f for f in list_of_files if f.suffix == ".mp4"]
+    print(len(list_of_files))
+    metadata_dict = {
+        "file_name": [],
+        "file_path": [],
+        "file_size": [],
+        "fps": [],
+        "frame_count": [],
+        "duration": [],
+        "width": [],
+        "height": [],
+        "num_pixels": [],
+    }
+
+    for video in tqdm.tqdm(list_of_files):
+        # extract metadata from the video
+        cap = cv2.VideoCapture(str(video))
+        metadata_dict["file_name"].append(video.name)
+        metadata_dict["file_path"].append(video)
+        metadata_dict["file_size"].append(video.stat().st_size)
+        metadata_dict["fps"].append(cap.get(cv2.CAP_PROP_FPS))
+        metadata_dict["frame_count"].append(int(cap.get(cv2.CAP_PROP_FRAME_COUNT)))
+        metadata_dict["duration"].append(
+            metadata_dict["frame_count"][-1] / metadata_dict["fps"][-1]
+        )
+        metadata_dict["width"].append(cap.get(cv2.CAP_PROP_FRAME_WIDTH))
+        metadata_dict["height"].append(cap.get(cv2.CAP_PROP_FRAME_HEIGHT))
+        metadata_dict["num_pixels"].append(
+            metadata_dict["width"][-1] * metadata_dict["height"][-1]
+        )
+        video_output_path = pathlib.Path(f"{output_dir}/{video.stem}").resolve()
+        video_output_path.mkdir(parents=True, exist_ok=True)
+
+        # load the video
+        cap = cv2.VideoCapture(str(video))
+        # iterate over the frames and save them as tiff files
+        for frame in range(int(cap.get(cv2.CAP_PROP_FRAME_COUNT))):
+            cap.set(cv2.CAP_PROP_FRAME_COUNT, frame)
+            ret, img = cap.read()
+            img_file_path = pathlib.Path(
+                f"{video_output_path}/{frame}_{video.stem}.tiff"
+            ).resolve()
+            if ret:
+                if not img_file_path.exists():
+                    tifffile.imwrite(
+                        img_file_path,
+                        img,
+                    )
+        cap.release()
+    # write the metadata to a csv file
+    metadata_df = pd.DataFrame(metadata_dict)
+    metadata_df.to_csv(output_dir / "metadata_of_tiffs.csv", index=False)
```

Comment:
For my understanding, why csv instead of parquet here?

Reply:
No real reason. Should be parquet, I will refactor to parquet now - good catch.